Lib/test/test_pyexpat.py

```diff
@@ -821,5 +823,107 @@ def start_element(name, _):
         self.assertEqual(started, ['doc'])
 
 
+class AttackProtectionTest(unittest.TestCase):
+
+    def exponential_expansion_payload(self, ncols, nrows, text='.'):
+        """Create a billion laughs attack payload.
+
+        Be careful: the number of total items is pow(n, k), thereby
+        requiring at least pow(ncols, nrows) * sizeof(text) memory!
+        """
+        # 'indent' affects the peak amplification factor and allocation
+        indent = ' ' * 2
+        body = textwrap.indent('\n'.join(
+            f'<!ENTITY row{i + 1} "{f"&row{i};" * ncols}">'
+            for i in range(nrows)
+        ), indent)
+        return f"""\
+<?xml version="1.0"?>
+<!DOCTYPE doc [
+{indent}<!ENTITY row0 "{text}">
+{indent}<!ELEMENT doc (#PCDATA)>
+{body}
+]>
+<doc>&row{nrows};</doc>
+"""
+
+    def test_set_alloc_tracker_maximum_amplification(self):
+        # On WASI, the maximum amplification factor of the payload may differ,
+        # so we craft a payload that is likely to yield an amplification factor
+        # way larger than 1.0 and way smaller than 10^5.
+        payload = self.exponential_expansion_payload(1, 2)
+
+        p = expat.ParserCreate()
+        # Unconditionally enable maximum amplification factor.
+        p.SetAllocTrackerActivationThreshold(0)
+        # Use a max amplification factor likely to be below the real one.
+        self.assertIsNone(p.SetAllocTrackerMaximumAmplification(1.0))
+        msg = r"out of memory: line \d+, column \d+"
+        self.assertRaisesRegex(expat.ExpatError, msg, p.Parse, payload, True)
+
+        # Re-create a parser as the current parser is now in an error state.
+        p = expat.ParserCreate()
+        # Unconditionally enable maximum amplification factor.
+        p.SetAllocTrackerActivationThreshold(0)
+        self.assertIsNone(p.SetAllocTrackerMaximumAmplification(10_000))
+        self.assertIsNotNone(p.Parse(payload, True))
+
+    def test_set_alloc_tracker_maximum_amplification_infty(self):
```

Comment:
```suggestion
    def test_set_alloc_tracker_maximum_amplification_infinity(self):
```
(just an idea, and because `git grep infty` did not return any matches)

Reply:
I'm too used to using `\infty` in LaTeX :)